runtime/convert.c

```diff
@@ -14,3 +15,35 @@ TaggedPtr _bal_convert_to_float(TaggedPtr tp) {
     }
     return floatToTagged(d);
 }
+
+TaggedIntWithOverflow _bal_convert_to_int(TaggedPtr tp) {
+    int64_t val;
+    bool overflowed;
+    TaggedIntWithOverflow res;
+    switch (getTag(tp) & UT_MASK) {
+        case TAG_FLOAT:
+            {
+                struct FloatToIntResult intRes = _bal_float_to_int(taggedToFloat(tp));
```

Comment:
Get rid of `struct FloatToIntResult` and use `IntWithOverflow` instead. (`FloatToIntResult` should be using `int64_t` not `uint64_t`).

Get rid of `overflowed` variable.

`TaggedIntWithOverflow` is misnamed because the tagged value isn't necessarily an `int`. Fix the name of the struct and the name of the member accordingly.

Reply:
> Get rid of `struct FloatToIntResult` and use `IntWithOverflow` instead. (`FloatToIntResult` should be using `int64_t` not `uint64_t`).
> 

@jclark, to be more clear about this suggestion, do you expect to change all places where `FloatToIntResult` is used into `IntWithOverflow`?

Reply:
Yes - we don't need both of them.

Reply:
You can do this in a separate clean-up PR if you prefer.

Reply:
@jclark, I will do it in a separate PR. 

What I have to do is,
1. Create struct like
```
typedef struct {
    TaggedPtr tp;
    bool overflowed;
} TaggedPtrWithOverflow;
```

2. Change `_bal_float_to_int` and `_bal_decimal_to_int` return above struct.
3. In number.bal if repr of reg is BASE_REPR_TAGGED  to store `tp`, otherwise store untagged value of `tp`.

Reply:
No, that's not right. First member of FloatToIntResult is not a TaggedPtr.

```
struct FloatToIntResult {
    uint64_t value;
    // LLVM type is i1, no equalent in C. i1 gets lowered to 8-bits.
    // So using bool (which is also 8-bit) works.
    bool overflowed;
};
```

You don't need to change _bal_decimal_with_int. You just replace `struct FloatToIntResult` with `IntWithOverflow`.

Reply:
Fixed in #803